setup.py

```diff
@@ -11,9 +11,30 @@
 SRC_DIR = os.path.join(THIS_DIR, "src")
 SNOWPARK_SRC_DIR = os.path.join(SRC_DIR, "snowflake", "snowpark")
 CONNECTOR_DEPENDENCY_VERSION = ">=3.2.0, <4.0.0"
+INSTALL_REQ_LIST = [
+    "setuptools>=40.6.0",
+    "wheel",
+    f"snowflake-connector-python{CONNECTOR_DEPENDENCY_VERSION}",
+    "pyyaml",
+]
+CLOUDPICKLE_REQ_LIST = ["cloudpickle>=1.6.0,<=2.0.0"]
+# this allows us to update the cloudpickle dependency by env var when building package privately
+# without re-releasing package just to update the dependency, example if we want to lower cloudpickle dep to 2.1.0:
```

Comment:
Just curious, why would we want to pin cloudpickle version in private builds?

Reply:
I did this mainly to pin the lower bound, cloudpickle >= 2.1.0 supports python 3.11, but right now we are testing the latest cloudpickle 2.2.1 in our channel, we don't have 2.1.0.

if not pin the lower bound, cloudpickle < 2.1.0 won't work with snowpark-python in py311 env